### DISCUSSION TITLE
How to build ts with tsc command for extensions?

### DISCUSSION THREAD

**Discussion by Enhed** [directus/directus#15992] [Archive]
After `"@directus/extensions-sdk": "9.16.1"` `tsc` generating invalid files for extensions, so its last working version for me now.  
I cant use native build for some reason.  
i get error: 
<img width="488" alt="image" src="https://user-images.githubusercontent.com/36981278/195680088-be83440c-21c8-40e4-9847-457e178873da.png">
and fiy: after message about `couldnt register hook` writes `loaded extensions`, but its not true

**Comment by br41nslug** [comment]
The error looks like an outdated version of `@directus/extension-sdk` or `@directus/shared` with a newer version of `nanoid` I believe they removed all support for cjs not to long ago resulting in these `require` errors. But it should be fixed in later version could you double check the versions of all used packages?

**Reply by Enhed** [reply]
if i update all packages to new version (or create new fresh directus project) then i will have this error. But if i install `@directus/extensions-sdk` with `9.16.1` all works.   
if I use the standard build mechanism, then everything works, but it does not suit me now, I would like to use the `tsc` mechanism to build extensions. Maybe there is a way to build files as `tsc`, but what would work like from a standard directus-extension build?

**Comment by br41nslug** [comment]
> if I use the standard build mechanism, then everything works

Ah thats good i thought it was the extension-sdk build that was failing for you.
> Maybe there is a way to build files as tsc, but what would work like from a standard directus-extension build?

Can't say i've tried building extensions using just `tsc` but could be possible for api extensions. From the error you've shown it looks like `tsc` is not dealing with "externalDependencies" properly. I did find this post with some possible workarounds you could try: https://stackoverflow.com/questions/33408643/typescript-bundling-not-working-for-external-modules
